augur/filter.py

```diff
@@ -18,12 +15,11 @@
 import re
 import sys
 from tempfile import NamedTemporaryFile
-import treetime.utils
-from textwrap import dedent
 from typing import Collection
 
 from .index import index_sequences, index_vcf
 from .io import open_file, read_metadata, read_sequences, write_sequences
+from .utils import date_parsing
```

Comment:
Zooming out a bit to the broader Augur context, I've thought for a while now that we should have all of our date-handling logic in its own top-level `dates.py` module. Although we tend to reserve top-level modules for subcommands (e.g., `align`, `refine`, `tree`, etc.), that's never been a hard convention (see `titers.py` and `titer_models.py` or `frequencies.py` and `frequency_estimators.py`). The `utils` module began as a dumping ground or catch-all for functions that didn't obviously belong to a single subcommand or that were used across those subcommands. The `io` module is a newer example of how those same types of functions could be logically grouped into their own module, though. Related to this historical context, the `util_support` directory was added as part of a burst of contributions by a single external developer who preferred to write in an object-oriented style. I don't think anyone internally would use this approach today but would instead prefer individual well-scoped modules with functions where possible.

For the work you've done here, I'd consider placing `numeric_date`, etc. in `dates.py` and then directly importing the functions/variables needed in each calling module. This line in `filter` would become:

```
from .dates import numeric_date, numeric_date_type, SUPPORTED_DATE_HELP_TEXT
```

This follows the same pattern you see for `index` and `io` in the lines above. This organization would more clearly designate `dates` as a top-level concern of Augur's.

Reply:
seconded

Reply:
💯 This opens up the door to consolidate other date parsing logic such as:

https://github.com/nextstrain/augur/blob/b464692c9681e32b1ff0d590d195a8ad21fbe882/augur/utils.py#L115-L131

https://github.com/nextstrain/augur/blob/b464692c9681e32b1ff0d590d195a8ad21fbe882/augur/utils.py#L133

https://github.com/nextstrain/augur/blob/b464692c9681e32b1ff0d590d195a8ad21fbe882/augur/utils.py#L78-L79

but let's do that in another PR.

Reply:
Seems reasonable to me.

Reply:
Done: ea17e7ba918badf6d404420d8a8e2286ad19b692...3413c81aeca0b99da8f4ed5b9c773226c95a3cd8